src/main/java/com/kryptnsotic/rhizome/converters/RhizomeConverter.java

```diff
@@ -0,0 +1,80 @@
+package com.kryptnsotic.rhizome.converters;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Type;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import retrofit.converter.ConversionException;
+import retrofit.converter.Converter;
+import retrofit.mime.TypedByteArray;
+import retrofit.mime.TypedInput;
+import retrofit.mime.TypedOutput;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.geekbeast.rhizome.registries.ObjectMapperRegistry;
+
+/**
+ * A {@link Converter} that handles both json and byte arrays. Based off work by Kai Waldron
+ * (kaiwaldron@gmail.com)
+ * 
+ * @author Matthew Tamayo-Rios &lt;matthew@kryptnostic.com&gt;
+ *
+ */
+public class RhizomeConverter implements Converter {
+    private static final String JSON_MIME_TYPE = "application/json; charset=UTF-8";
+    private static final String BYTE_MIME_TYPE = "application/octet-stream";
+    private static final Logger logger         = LoggerFactory.getLogger( RhizomeConverter.class );
+    private final ObjectMapper  objectMapper;
+
+    public RhizomeConverter() {
+        this.objectMapper = ObjectMapperRegistry.getPlainMapper();
+    }
+
+
+    @Override
+    public Object fromBody( TypedInput body, Type type ) throws ConversionException {
+        try {
+            if ( StringUtils.equals( body.mimeType(), BYTE_MIME_TYPE ) ) {
+                return IOUtils.toByteArray( body.in() );
+            }
+
+            JavaType javaType = objectMapper.getTypeFactory().constructType( type );
+
+            InputStream in = body.in();
+
+            if ( in.available() == 0 ) {
```

Comment:
I have no idea :-/ There's not really anything to check other than that even though the implementation class in memory has all kinds of useful stuff in it. The javadoc for it makes things even more confusing:
     * Note that while some implementations of {@code InputStream} will return
     * the total number of bytes in the stream, many will not.  It is
     * never correct to use the return value of this method to allocate
     * a buffer intended to hold all data in this stream.

Soooo I think we should just remove the check and it will throw an IOException and we can return null in the catch.